src/utils/gitUtils.ts

```diff
@@ -0,0 +1,145 @@
+        throw new Error(localize('emptyWorkspace', 'Cannot create a Static Web App with an empty workspace.'));
+    }
+
+    if (!gitWorkspaceState.repo) {
+        const gitRequired: string = localize('gitRequired', 'A GitHub repository is required to create a Static Web App. Would you like to initialize your project as a git repository and create a GitHub remote?');
+        context.telemetry.properties.cancelStep = 'initRepo';
+
+        await ext.ui.showWarningMessage(gitRequired, { modal: true }, DialogResponses.yes);
+        const gitApi: API = await getGitApi();
+        let newRepo: Repository | null = await gitApi.init(uri);
+        if (!newRepo) {
+            const localGit: git.SimpleGit = git(uri.fsPath);
+            await localGit.init({});
+            newRepo = await gitApi.openRepository(uri);
+            if (!newRepo) {
+                throw new Error(localize('gitInitFailed', 'Git initialization failed.  Please initialize a git repository manually and attempt to create again.'));
+            }
+
+        }
+
+        await promptForCommit(newRepo, localize('initCommit', 'Initial commit'));
+        gitWorkspaceState.repo = newRepo;
+
+    } else if (!!gitWorkspaceState.remoteUrl && !gitWorkspaceState.hasAdminAccess) {
+        context.telemetry.properties.cancelStep = 'adminAccess';
+
+        const adminAccess: string = localize('adminAccess', 'Admin access to the GitHub repository is required.  Please use a repo with admin access or create a fork.');
+        await ext.ui.showWarningMessage(adminAccess, { modal: true });
+
+    } else if (gitWorkspaceState.dirty && gitWorkspaceState.repo) {
+        context.telemetry.properties.cancelStep = 'dirtyWorkspace';
+
+        const commitChanges: string = localize('commitChanges', 'Commit all working changes to create a Static Web App.');
+        await ext.ui.showWarningMessage(commitChanges, { modal: true }, { title: localize('commit', 'Commit') });
+        await promptForCommit(gitWorkspaceState.repo, localize('commitMade', 'Commit made from VS Code Azure Static Web Apps'));
+        gitWorkspaceState.dirty = false;
+    }
+
+    return <VerifiedGitWorkspaceState>gitWorkspaceState;
+}
+
+export async function promptForCommit(repo: Repository, value?: string): Promise<void> {
+    const commitPrompt: string = localize('commitPrompt', 'Enter a commit message.');
+    const commitOptions: CommitOptions = { all: true };
+
+    const commitMsg: string = await ext.ui.showInputBox({ prompt: commitPrompt, placeHolder: `${commitPrompt}..`, value });
+    try {
+        await repo.commit(commitMsg, commitOptions);
+    } catch (err) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        if (!/nothing to commit/.test(err.stdout)) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+            console.debug(parseError(err));
```

Comment:
I didn't think that console logs worked from our extensions. Was this just for debugging?